rust/arrow/src/memory.rs

```diff
@@ -38,11 +38,19 @@ pub fn free_aligned(p: *mut u8, size: usize) {
 
 pub fn reallocate(ptr: *mut u8, old_size: usize, new_size: usize) -> *mut u8 {
     unsafe {
-        std::alloc::realloc(
+        let new_ptr = std::alloc::realloc(
             ptr,
             Layout::from_size_align_unchecked(old_size, ALIGNMENT),
             new_size,
-        )
+        );
+        if new_size > old_size {
+            std::ptr::write_bytes(
```

Comment:
Fwiw, using the [`.write_bytes()`](https://doc.rust-lang.org/std/primitive.pointer.html#method.write_bytes) method on the pointer would be a bit more concise.